File: 프로그래머스/0/120854. 배열 원소의 길이/배열 원소의 길이.py
def solution(strlist):
    answer = []
    
    for str in strlist:
        answer.append(len(str))
    
    return answer

# for 변수 in 콜렉션: 파이썬 반복문
# 배열.append(): 배열에 값 추가
# len(): 길이 반환

# 다른 풀이 1
return list(map(len, strlist))
# map(함수, Iterable 객체) : 각 요소에 대해 함수를 적용하고 그 결과를 새로운 이터레이터로 반환
# list(): 이터레이터 -> 리스트로 변환

# 다른 풀이 2
return [len(str) for str in strlist]
# 리스트 컴프리헨션 : 간단한 반복 작업을 통해 새로운 리스트를 생성하는 파이썬의 기능
# [표현식 for 요소 in Iterable 객체]

File: 프로그래머스/0/120854. 배열 원소의 길이/배열 원소의 길이.java
class Solution {
    public int[] solution(String[] strlist) {
        int size = strlist.length;
        int[] answer = new int[size];
        
        for(int i = 0; i < size; i++) {
            answer[i] = strlist[i].length();
        }
        
        return answer;
    }
}

// 다른 풀이
return Arrays.stream(strList).mapToInt(String::length).toArray();
// Arrays.stream(strList): strList를 스트림으로 변환
/* .mapToInt(String::length)
    - mapToInt(): 스트림의 각 요소를 특정한 방식으로 변환 후 정수 스트림 생성

    - 메소드 레퍼런스 (클래스::메소드) : 람다식과 같은 기능, 다만 메소드를 참조하여 호출
    -- String::length == String객체의 length() 호출
*/
